.github/workflows/checker.yml

```diff
@@ -0,0 +1,22 @@
+name: Checker
+on: [push, pull_request]
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Go 1.17
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17
+
+      - name: Check out source code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+
+      - name: Install checker tool
+        run: go install github.com/terakilobyte/checker@latest
+
+      - name: Run checker tool
+        run: git diff --name-only HEAD^..HEAD | tr "\n" "," | xargs checker --changes
```

Comment:
Aside from the issue of other directives or their targets changing external to the repo, I wonder if this misses any links that might be change by a modification to snooty.toml source constants. Does it take too much time to run if you leave out this optimization?

Reply:
On large repos it can take a long time. For example, `docs` has ~9.2K links to check. The other issue is, and the next major improvement, would be to introduce more sophisticated throttling like [muffet](https://github.com/raviqqe/muffet)

Reply:
Agreed it was ok in an offline discussion.
One additional point: if you set this up to run on every commit to every repo, you'll potentially get a lot of duplicate checking. While I don't think there's a way to conditionally trigger github actions, maybe you exit unless the latest commit message contains something like "check links". Open to other solutions too.